### PR TITLE
feat(cli): options --skip-warehouse-catalog and --use-dbt-list

### PR DESCRIPTION
**Pull request by owlas**
Closes: 

### Description:

Main goal: 
Deploy/preview faster. Main time consumers is `dbt compile` command that generates the manifest.json and fetching the warehouse catalog to apply the correct dimension types.

Changes:
- new `--use-dbt-list` option - uses `dbt list` instead fo `dbt compile` which is much faster
- new `--skip-warehouse-catalog` option - assumes YML has the correct dimension types and skips fetching the warehouse catalog

Note: Commands `preview` and `deploy --create` still need warehouse credentials to create the project.
Note2: `--use-dbt-list` option is not supported when using dbt 1.3

Tests:

`lightdash deploy` ( backwards compatibility )

Screenshot 2024-01-12 at 13 27 42

`lightdash deploy --skip-warehouse-catalog`

Screenshot 2024-01-12 at 13 32 10

`lightdash deploy --use-dbt-list`

Screenshot 2024-01-12 at 13 33 54

`lightdash deploy --use-dbt-list --skip-warehouse-catalog`

Screenshot 2024-01-12 at 13 32 55

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/cli/src/handlers/compile.ts [hunk 1]
```diff
@@ -97,7 +98,9 @@ export const compile = async (options: CompileHandlerOptions) => {
             : undefined,
     });
     const manifest = await loadManifest({ targetDir: context.targetDir });
-    const models = getModelsFromManifest(manifest);
+    const models = getModelsFromManifest(manifest).filter((model) =>
+        modelIds.includes(model.unique_id),
```
Comment (ZeRego) [inline packages/cli/src/handlers/compile.ts:102]: fix case when dbt compile was skipped and modelIds is empty

File: packages/cli/src/handlers/compile.ts [hunk 2]
```diff
@@ -119,9 +122,10 @@ ${errors.join('')}`),
     }
 
     // Ideally we'd skip this potentially expensive step
-    const catalog = await warehouseClient.getCatalog(
-        getSchemaStructureFromDbtModels(validModels),
-    );
+    // const catalog = await warehouseClient.getCatalog(
+    //     getSchemaStructureFromDbtModels(validModels),
+    // );
+    const catalog = {};
```
Comment (ZeRego) [inline packages/cli/src/handlers/compile.ts:128]: TODO: add a param to skip this `--skip-warehouse-catalog`

File: packages/cli/src/handlers/dbt/compile.ts [hunk 3]
```diff
@@ -53,13 +53,27 @@ const optionsToArgs = (options: DbtCompileOptions): string[] =>
...
+            '--output-keys',
+            'unique_id',
+        ];
+        GlobalState.debug(`> Running: dbt ls ${args.join(' ')}`);
+        const { stdout, stderr } = await execa('dbt', ['ls', ...args]);
+        const models = stdout
+            .split('\n')
+            .slice(1)
+            .map((line) => JSON.parse(line).unique_id);
+        console.error(models);
```
Comment (ZeRego) [inline packages/cli/src/handlers/dbt/compile.ts:74]: TODO: use GlobalState.debug

File: packages/common/src/compiler/translator.ts [hunk 4]
```diff
@@ -253,9 +253,9 @@ export const convertTable = (
     dbtMetrics: DbtMetric[],
     startOfWeek?: WeekDay | null,
 ): Omit<Table, 'lineageGraph'> => {
-    if (!model.compiled) {
-        throw new NonCompiledModelError(`Model has not been compiled by dbt`);
-    }
+    // if (!model.compiled) {
+    //     throw new NonCompiledModelError(`Model has not been compiled by dbt`);
+    // }
```
Comment (ZeRego) [inline packages/common/src/compiler/translator.ts:258]: TODO: can we remove this without changing the BE to also use `ld`?

**Comment by ZeRego** [comment]
We should keep the dbt compile for now and introduce `--experimental-dbt-list`

File: packages/cli/src/dbt/validation.ts [hunk 5]
```diff
@@ -31,10 +31,6 @@ export const validateDbtModel = async (
     const validator = new ManifestValidator(manifestVersion);
     const results = models.reduce<DbtModelsGroupedByState>(
         (acc, model) => {
-            if (model.compiled === undefined) {
```
Comment (owlas) [inline packages/cli/src/dbt/validation.ts:34]: @ZeRego do you remember what this was actually checking for?

Comment (ZeRego) [reply packages/cli/src/dbt/validation.ts:34]: would skip models that weren't compiled. eg: `lightdash compile -s tag:lightdash`

**Review by ZeRego — APPROVED** [review]

**Pull request merged**